WalletWasabi/Tor/Control/TorControlClient.cs

```diff
@@ -30,11 +35,50 @@ internal TorControlClient(PipeReader pipeReader, PipeWriter pipeWriter)
 			TcpClient = null;
 			PipeReader = pipeReader;
 			PipeWriter = pipeWriter;
+			ReaderCts = new();
+
+			SyncChannel = Channel.CreateUnbounded<TorControlReply>(Options);
+			AsyncChannels = new List<Channel<TorControlReply>>();
+			ReaderLoopTask = Task.Run(ReaderLoopAsync);
```

Comment:
Why `Task.Run` here?

Reply:
This is because `ReaderLoopAsync` is designed to run as a separate task which reads incoming messages from Tor Control. This is needed because Tor Control can send an event at any time and we need to process them.

Reply:
I understand that. My question is can't we just `ReaderLoopTask = ReaderLoopAsync();` (without awaiting) and have the same behavior?

Reply:
Ah, it simply did not occur to me. It should be probably as you say.

Reply:
Anyhow, I think that gives us a warning, so the current version results in more readable code.